http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/AutopauseMonitor.scala

```diff
@@ -25,84 +24,87 @@ class AutopauseMonitor[F[_]](
+                )
+                .as(true),
+            metrics.incrementCounter("autoPause/activeKernelClusters") >>
+              logger
+                .info(
+                  s"Not going to auto pause cluster ${cluster.cloudContext}/${cluster.runtimeName} due to active kernels"
+                )
+                .as(false)
+          )
+        } else F.pure(isIdle)
+    }
 
-              maxKernelActiveTimeExceeded.ifM(
-                metrics.incrementCounter("autoPause/maxKernelActiveTimeExceeded") >>
-                  logger
-                    .info(
-                      s"Auto pausing ${cluster.cloudContext}/${cluster.runtimeName} due to exceeded max kernel active time"
-                    )
-                    .as(true),
-                metrics.incrementCounter("autoPause/activeKernelClusters") >>
-                  logger
-                    .info(
-                      s"Not going to auto pause cluster ${cluster.cloudContext}/${cluster.runtimeName} due to active kernels"
-                    )
-                    .as(false)
-              )
-            } else F.pure(isIdle)
-        }
-      }
-      _ <- metrics.gauge("autoPause/numOfRuntimes", pauseableClusters.length)
-      _ <- pauseableClusters.traverse_ { cl =>
-        val traceId = TraceId(s"fromAutopause_${UUID.randomUUID().toString}")
-        for {
-          _ <- clusterQuery.updateClusterStatus(cl.id, RuntimeStatus.PreStopping, now).transaction
-          _ <- metrics.incrementCounter("autoPause/pauseRuntimeCounter")
-          _ <- logger.info(Map("traceId" -> traceId.asString))(s"Auto freezing runtime ${cl.projectNameString}")
-          _ <- publisherQueue.offer(LeoPubsubMessage.StopRuntimeMessage(cl.id, Some(traceId)))
-        } yield ()
-      }
+  override def dbSource(): F[Seq[RuntimeToAutoPause]] =
+    clusterQuery.getClustersReadyToAutoFreeze.transaction
+
+  override def action(a: RuntimeToAutoPause, traceId: TraceId, now: Instant)(implicit F: Async[F]): F[Unit] =
+    for {
+      _ <- clusterQuery.updateClusterStatus(a.id, RuntimeStatus.PreStopping, now).transaction
+      _ <- publisherQueue.offer(LeoPubsubMessage.StopRuntimeMessage(a.id, Some(traceId)))
     } yield ()
 }
 
 object AutopauseMonitor {
-  def apply[F[_]](config: AutoFreezeConfig, jupyterDAO: JupyterDAO[F], publisherQueue: Queue[F, LeoPubsubMessage])(
+  def process[F[_]](config: AutoFreezeConfig, jupyterDAO: JupyterDAO[F], publisherQueue: Queue[F, LeoPubsubMessage])(
     implicit
+    dbRef: DbReference[F],
+    ec: ExecutionContext,
     F: Async[F],
-    metrics: OpenTelemetryMetrics[F],
-    logger: StructuredLogger[F],
+    openTelemetry: OpenTelemetryMetrics[F],
+    logger: StructuredLogger[F]
+  ): Stream[F, Unit] = {
+    val autopauseMonitor = apply(config, jupyterDAO, publisherQueue)
+
+    implicit val runtimeToAutoPauseShowInstance: Show[RuntimeToAutoPause] =
+      Show[RuntimeToAutoPause](runtimeToAutoPause => runtimeToAutoPause.projectNameString)
```

Comment:
What is a `Show`?

Reply:
https://typelevel.org/cats/typeclasses/show.html

Reply:
Thank you! That makes sense :)

Is this something that could move into the `RuntimeToAutopause` definition? Seems like it's basically a better `toString`; it would make sense to have that all encapsulated in one object

Reply:
that wouldn't work becuz `A` is generic....https://github.com/DataBiosphere/leonardo/pull/3915/files#diff-84015af49ce0da0253bc5974a8fbe0b8dac33f7871008ad4e787386a3fdeb722R74 is where `show` is used

Reply:
`A` should be restricted to some degree though, yes? For example currently the only `A` we use is `RuntimeToAutoPause`; it might make sense to have that extend or implement a contract `MonitorCandidate` which could hold the show logic. Then `A <: MonitorCandidate` would both make the code self-documenting and allow the show logic to live where it belongs, with the item it's showing

Reply:
no. if we don't have to restrict anything here, we should leave it generic...

Reply:
The proliferation of generic types makes our legacy code really hard to read, in combination with implicits. I don't understand why this refactor requires relaxing the type-safety of this method from one type (RuntimeToAutopause) to Any

Reply:
`A` is different from `Any`..everything will still type safe at compile time

Reply:
In that case, would you add comments to the scaladoc describing what A is intended to represent?